torchgeo/datasets/substation_seg.py

```diff
@@ -0,0 +1,170 @@
+
+        Returns:
+            A dictionary containing the image and corresponding mask.
+        """
+        image_filename = self.image_filenames[index]
+        image_path = os.path.join(self.image_dir, image_filename)
+        mask_path = os.path.join(self.mask_dir, image_filename)
+
+        image = np.load(image_path)['arr_0']
+        # standardizing image
+        if self.normalizing_type == 'percentile':
+            image = (
+                image - self.normalizing_factor[:, 0].reshape((-1, 1, 1))
+            ) / self.normalizing_factor[:, 2].reshape((-1, 1, 1))
+        elif self.normalizing_type == 'zscore':
+            # means = np.array([1431, 1233, 1209, 1192, 1448, 2238, 2609, 2537, 2828, 884, 20, 2226, 1537]).reshape(-1, 1, 1)
+            # stds = np.array([157, 254, 290, 420, 363, 457, 575, 606, 630, 156, 3, 554, 523]).reshape(-1, 1, 1)
+            image = (image - self.args.means) / self.args.stds
+        else:
+            image = image / self.normalizing_factor
+            # clipping image to 0,1 range
+            image = np.clip(image, 0, 1)
+
+        # selecting channels
+        if self.in_channels == 3:
+            image = image[:, [3, 2, 1], :, :]
+        else:
+            if self.model_type == 'swin':
+                image = image[
+                    :, [3, 2, 1, 4, 5, 6, 7, 10, 11], :, :
+                ]  # swin only takes 9 channels
+            else:
+                image = image[:, : self.in_channels, :, :]
+
+        # handling multiple images across timepoints
+        if self.use_timepoints:
+            image = image[:4, :, :, :]
+            if self.args.timepoint_aggregation == 'concat':
+                image = np.reshape(
+                    image, (-1, image.shape[2], image.shape[3])
+                )  # (4*channels,h,w)
+            elif self.args.timepoint_aggregation == 'median':
+                image = np.median(image, axis=0)
+        else:
+            # image = np.median(image, axis=0)
+            # image = image[0]
+            if self.args.timepoint_aggregation == 'first':
+                image = image[0]
+            elif self.args.timepoint_aggregation == 'random':
+                image = image[np.random.randint(image.shape[0])]
+
+        mask = np.load(mask_path)['arr_0']
+        mask[mask != 3] = 0
+        mask[mask == 3] = 1
+
+        image = torch.from_numpy(image)
+        mask = torch.from_numpy(mask).float()
+        mask = mask.unsqueeze(dim=0)
+
+        if self.mask_2d:
+            mask_0 = 1.0 - mask
+            mask = torch.concat([mask_0, mask], dim=0)
+
+        if self.image_resize:
+            image = self.image_resize(image)
+
+        if self.mask_resize:
+            mask = self.mask_resize(mask)
+
+        return {'image': image, 'mask': mask}
+
+    def __len__(self) -> int:
+        """Returns the number of items in the dataset."""
+        return len(self.image_filenames)
+
+    def plot(self) -> None:
```

Comment:
The plot method takes a sample as input and plots it. See every other dataset for an example of this.